### PR TITLE
[t3]-search controller test

### PR DESCRIPTION
**Pull request by KenXiong123**
Goal
Implement unit test and refactor logics for Search API.

In this PR

Implement SearchControllerTest.
Refactor SearchController and its unit test.
All unit tests passed.

### CODE REVIEW

File: src/main/java/com/lgtm/easymoney/controllers/SearchController.java [hunk 1]
```diff
@@ -25,8 +25,13 @@ public SearchController(SearchService searchService) {
...
   @GetMapping("/id/{id}")
   @Operation(summary = "Method to retrieve the public profile of a user by user ID.")
-  public ResponseEntity<SearchRsp> searchById(@PathVariable(value = "id") String id) {
-    return searchService.searchById(Long.valueOf(id));
+  public ResponseEntity<SearchRsp> searchById(@PathVariable(value = "id") String id) throws Exception {
+    if (!id.isEmpty()){
+      Long uid = Long.parseLong(id);
+      return searchService.searchById(uid);
+    }else{
+      throw new Exception("User ID can't be empty.");
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/controllers/SearchController.java:33]: use our own ex handler?

Comment (howieraem) [reply src/main/java/com/lgtm/easymoney/controllers/SearchController.java:33]: You can just use @PathVariable(...) Long id, and use @NotNull there as well. See https://stackoverflow.com/questions/23325413/what-runtimeexception-can-spring-crudrepository-throw

File: src/test/java/com/lgtm/easymoney/controllers/SearchControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                jsonPath("$.userProfiles[0].accountName").value(accountName),
+                jsonPath("$.userProfiles[0].address").value(address),
+                jsonPath("$.userProfiles[0].email").value(email),
+                jsonPath("$.userProfiles[0].phone").value(phone),
+                jsonPath("$.userProfiles[0].userType").value(String.valueOf(userType)));
+    }
+
+    @Test
+    public void searchByIdFailedWithInvalidPathVar() throws Exception {
+        Mockito.doThrow(new NumberFormatException(null))
```
Comment (rl3250) [inline src/test/java/com/lgtm/easymoney/controllers/SearchControllerTest.java:169]: where's the act and assert?

**Review by howieraem — CHANGES_REQUESTED** [review]

File: src/test/java/com/lgtm/easymoney/controllers/SearchControllerTest.java (hunk 2 above)
Comment (howieraem) [reply src/test/java/com/lgtm/easymoney/controllers/SearchControllerTest.java:169]: Example: https://github.com/howieraem/COMS4156-ASE/blob/e4035f78c14ebfe3f8a738f1fd8f216f3113c563/src/test/java/com/lgtm/easymoney/controllers/GroupControllerTest.java#L228

**Review by howieraem — APPROVED** [review]

**Pull request merged**